conans/test/functional/package_id/package_id_requires_modes_test.py

```diff
@@ -592,3 +592,35 @@ def package_revision_mode_editable_test(self):
         client2.save({"conanfile.py": GenConanfile().with_require_plain("dep2/1.0@user/testing")})
         client2.run('create . consumer/1.0@user/testing --build')
         self.assertIn("consumer/1.0@user/testing: Created", client2.out)
+
+
+class PackageRevisionModeTestCase(unittest.TestCase):
+
+    def test_transtive_package_revision_mode(self):
+        t = TestClient()
+        t.save({
+            'package1.py': GenConanfile("pkg1"),
+            'package2.py': GenConanfile("pkg2").with_require_plain("pkg1/1.0"),
+            'package3.py': textwrap.dedent("""
+                from conans import ConanFile
+                class Recipe(ConanFile):
+                    requires = "pkg2/1.0"
+                    def package_id(self):
+                        self.info.requires["pkg1"].package_revision_mode()
+            """)
+        })
+        t.run("create package1.py pkg1/1.0@")
+        t.run("create package2.py pkg2/1.0@")
+
+        # If we only build pkg1, we get a new packageID for pkg3
+        t.run("create package3.py pkg3/1.0@ --build=pkg1", assert_error=True)
```

Comment:
In @jgsogo this didn't fail. I have added a ``assert_error``, this should fail, because missing binary for ``pkg3``, as the logic of ``--build`` in the ``create`` command require to define ``--build=pkg3`` to actually build the moment that any ``--build`` is used.

Reply:
I didn't realize it! This is fixing a current bug and changing the behavior. Previously we were computing the new package-ID, but the output was `pkg3/1.0: Already installed!`